### PR TITLE
added event grid namespace system events

### PR DESCRIPTION
**Pull request by george-guirguis**
# Data Plane API - Pull Request

## API Info: The Basics
Most of the information about your service should be captured in the issue that serves as your [*API Spec engagement record*](https://aka.ms/azsdk/onboarding/restapischedule).

* Link to API Spec engagement record issue:

Is this review for (select one):

- [ ] a private preview
- [ ] a public preview
- [ ] GA release

### Change Scope
<sup>This section will help us focus on the specific parts of your API that are new or have been modified. <br/>Please share a link to the design document for the new APIs, a link to the previous API Spec document (if applicable), and the root paths that have been updated. </sup>
* Design Document:
* Previous API Spec Doc:
* Updated paths:

## ❔Got questions? Need additional info?? We are here to help!

<details>
  <summary> Contact us!</summary>

The [Azure API Review Board](https://aka.ms/azsdk/onboarding/restapischedule) is dedicated to helping you create amazing APIs. You can read about our mission and learn more about our process on our [wiki](https://aka.ms/azsdk/onboarding/restapischedule).
* 💬 [Teams Channel](https://teams.microsoft.com/l/channel/19%3a3ebb18fded0e47938f998e196a52952f%40thread.tacv2/General?groupId=1a10b50c-e870-4fe0-8483-bf5542a8d2d8&tenantId=72f988bf-86f1-41af-91ab-2d7cd011db47)
* 💌 [email](mailto://azureapirbcore@microsoft.com)

</details>

<details>
  <summary>Click here for links to tools, specs, guidelines & other good stuff</summary>

### Tooling
 * [Open API validation tools](https://aka.ms/swaggertools) were run on this PR. Go here to see [how to fix errors](https://aka.ms/ci-fix)
 * [Spectral Linting](https://aka.ms/style)
 * [Open API Hub](https://aka.ms/openapiportal)

### Guidelines & Specifications
 * [Azure REST API Guidelines](https://aka.ms/guidelines)
 * [OpenAPI Style Guidelines](https://aka.ms/style)
 * [Azure Breaking Change Policy](https://aka.ms/AzBreakingChangesPolicy)

### Helpful Links
 * [Azure DevTools Wiki](https://aka.ms/azapi)

</details>

### CODE REVIEW

File: specification/eventgrid/data-plane/Microsoft.EventGrid/preview/2023-06-01-preview/EventGrid.json [hunk 1]
```diff
@@ -672,6 +672,146 @@
...
+        },
+        "createdOn": {
+          "type": "string",
+          "description": "Time the client resource is created based on the provider's UTC time."
+        },
+        "updatedOn": {
+          "type": "string",
+          "description": "Time the client resource is last updated based on the provider's UTC time. If the client resource was never updated, this value is identical to the value of the 'createdOn' property."
+        },
+        "attributes": {
```
Comment (JoshLove-msft) [inline specification/eventgrid/data-plane/Microsoft.EventGrid/preview/2023-06-01-preview/EventGrid.json:744]: Is this supposed to be a dictionary? You would need to add additionalProperties along with the type of the value, e.g. https://github.com/Azure/azure-rest-api-specs/blob/main/specification/eventgrid/data-plane/Microsoft.Media/stable/2018-01-01/MediaServices.json#L114-L116

File: specification/eventgrid/data-plane/Microsoft.EventGrid/preview/2023-06-01-preview/EventGrid.json [hunk 2]
```diff
@@ -672,6 +672,146 @@
...
+        },
+        "clientName": {
+          "type": "string",
+          "description": "Name of the client resource in the Event Grid namespace."
+        },
+        "namespaceName": {
+          "type": "string",
+          "description": "Name of the Event Grid namespace where the MQTT client was created or updated."
+        },
+        "state": {
```
Comment (JoshLove-msft) [inline specification/eventgrid/data-plane/Microsoft.EventGrid/preview/2023-06-01-preview/EventGrid.json:732]: This should be an extensible enum with the known values specified. Here is an example - https://github.com/l0lawrence/azure-rest-api-specs/blob/egevents_restructure/specification/eventgrid/data-plane/Microsoft.Communication/stable/2018-01-01/AzureCommunicationServices.json#L497-L507

File: specification/eventgrid/data-plane/Microsoft.EventGrid/preview/2023-06-01-preview/EventGrid.json [hunk 3]
```diff
@@ -672,6 +672,146 @@
...
+          "type": "string",
+          "description": "Name of the Event Grid namespace where the MQTT client was connected"
+        },
+        "sequenceNumber": {
+          "type": "string",
+          "description": "A number that helps indicate order of MQTT client session connected or disconnected events. Latest event will have a sequence number that is higher than the previous event."
+        }
+      }
+    },
+    "MQTTClientSessionDisconnectedEventProperties": {
```
Comment (JoshLove-msft) [inline specification/eventgrid/data-plane/Microsoft.EventGrid/preview/2023-06-01-preview/EventGrid.json:790]: Why do we define separate properties definitions for each type? If the goal is to reduce duplication we should create a base definition that has the common properties and add the additional properties in each derived definition.

File: specification/eventgrid/data-plane/Microsoft.EventGrid/preview/2023-06-01-preview/EventGrid.json [hunk 4]
```diff
@@ -672,6 +672,146 @@
...
+        "clientSessionName": {
+          "type": "string",
+          "description": "Unique identifier for the MQTT client's session. This case-sensitive string can be up to 128 characters long, and supports UTF-8 characters."
+        },
+        "namespaceName": {
+          "type": "string",
+          "description": "Name of the Event Grid namespace where the MQTT client was connected"
+        },
+        "sequenceNumber": {
+          "type": "string",
```
Comment (JoshLove-msft) [inline specification/eventgrid/data-plane/Microsoft.EventGrid/preview/2023-06-01-preview/EventGrid.json:785]: Shouldn't this be a number?

**Comment by l0lawrence** [comment]
Can we move these events and the SubscriptionValidation events under a new file called EventGridSystemEvents or something alike? @JoshLove-msft

**Comment by JoshLove-msft** [comment]
PR with proposed changes - https://github.com/Azure/azure-rest-api-specs/pull/25526

File: specification/eventgrid/data-plane/Microsoft.EventGrid/preview/2023-06-01-preview/EventGrid.json (hunk 1 above)
Comment (george-guirguis) [reply specification/eventgrid/data-plane/Microsoft.EventGrid/preview/2023-06-01-preview/EventGrid.json:744]: not sure what you mean with dictionary here.

File: specification/eventgrid/data-plane/Microsoft.EventGrid/preview/2023-06-01-preview/EventGrid.json (hunk 4 above)
Comment (george-guirguis) [reply specification/eventgrid/data-plane/Microsoft.EventGrid/preview/2023-06-01-preview/EventGrid.json:785]: that's correct!

File: specification/eventgrid/data-plane/Microsoft.EventGrid/preview/2023-06-01-preview/EventGrid.json (hunk 3 above)
Comment (george-guirguis) [reply specification/eventgrid/data-plane/Microsoft.EventGrid/preview/2023-06-01-preview/EventGrid.json:790]: I wasn't sure about the best practice/standard here. We can definitely follow the model you have in your pr

**Comment by JoshLove-msft** [comment]
Closing this out as https://github.com/Azure/azure-rest-api-specs/pull/25526 has been merged.

**Pull request closed without merging**